### PR TITLE
Implement show_empty functionality for the markdown card

### PR DESCRIPTION
**Pull request by illuzn**
## Proposed change

This implements `show_empty: true/ false` functionality for the markdown card. The functionality is similar to the _entity filter_ card.

This resolves a requested feature here (https://www.reddit.com/r/homeassistant/comments/y2nwz0/hide_markdown_card_if_empty/) for which there are no appropriate solutions.

It is possible using templates to create conditions which far exceed the capability of the conditional card or the "visibility" settings of cards so this functionality is required. For example, you currently cannot toggle visibility based on an attribute of a card currently.

The workaround suggested in that reddit thread (using a template sensor and outputting to state) isn't appropriate because the text output from a markdown card can exceed the maximum of 255 characters. 

There is no existing translation in the entity filter card for "show_empty" so the visual editor couldn't be implmented in this case.

By default, the existing behaviour is kept i.e. an empty card is shown which results in a small empty box.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
type: markdown
content: '{{ "Hello World" if 1 == 2 }}'
show_empty: false
```

## Additional information

- This PR fixes or closes issue: N/A
- This PR is related to issue or discussion: N/A
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/33720

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [X] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [X] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

## Summary by CodeRabbit

- **New Features**
  - Enhanced Markdown card configuration with a new `show_empty` option to control card visibility when there are no results.
  - Improved handling of empty result scenarios for better user experience with Markdown cards, ensuring they are hidden if no content is available.

### CODE REVIEW

**Comment by illuzn** [comment]
@coderabbitai review

**Comment by illuzn** [comment]
Linting error arises from hui-markdown-card.ts#L92:
https://github.com/illuzn/frontend/blob/5dd82ca74537b93b2ff68cb73b986dadc5b44fcf/src/panels/lovelace/cards/hui-markdown-card.ts#L89-L95
`Type 'string' is not assignable to '"error" | "info" | "warning" | "success"'`

I did not make this change but it seems to be a typing error.

**Comment by silamon** [comment]
> Linting error arises from hui-markdown-card.ts#L92: https://github.com/illuzn/frontend/blob/5dd82ca74537b93b2ff68cb73b986dadc5b44fcf/src/panels/lovelace/cards/hui-markdown-card.ts#L89-L95 `Type 'string' is not assignable to '"error" | "info" | "warning" | "success"'`
> 
> I did not make this change but it seems to be a typing error.

It's a warning which can be ignored and not introduced in your PR. 
Just run prettier (`yarn run format:prettier`) to get the CI green.

File: src/panels/lovelace/cards/hui-markdown-card.ts [hunk 1]
```diff
@@ -113,7 +114,18 @@ export class HuiMarkdownCard extends LitElement implements LovelaceCard {
     if (changedProps.has("_config")) {
       this._tryConnect();
     }
-
+    if (
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-markdown-card.ts:117]: We should check if `_templateResult` or `_config` changed first

File: src/panels/lovelace/cards/hui-markdown-card.ts [hunk 2]
```diff
@@ -113,7 +114,18 @@ export class HuiMarkdownCard extends LitElement implements LovelaceCard {
...
+      this._templateResult &&
+      this._templateResult.result.length === 0 &&
+      this._config.show_empty === false
+    ) {
+      if (!this.hidden) {
+        this.style.display = "none";
+        this.toggleAttribute("hidden", true);
+        fireEvent(this, "card-visibility-changed", { value: false });
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-markdown-card.ts:125]: You implemented logic to hide it, but there is no logic that unhides it ones the template does give a result?

File: src/panels/lovelace/cards/hui-markdown-card.ts [hunk 3]
```diff
@@ -114,17 +114,16 @@ export class HuiMarkdownCard extends LitElement implements LovelaceCard {
...
-      if (!this.hidden) {
+    if (this._templateResult && this._config.show_empty === false) {
+      if (this._templateResult.result.length === 0 && !this.hidden) {
         this.style.display = "none";
         this.toggleAttribute("hidden", true);
         fireEvent(this, "card-visibility-changed", { value: false });
+      } else {
+        this.style.display = "block";
+        this.toggleAttribute("hidden", false);
+        fireEvent(this, "card-visibility-changed", { value: true });
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-markdown-card.ts:125]: This means we will now do this every time the result is not empty, we should only do this is the card is hidden and the response not empty.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**